frontend/src/components/table/ServerPaginatedDataTable.tsx

```diff
@@ -0,0 +1,198 @@
+import {
+  component$,
+  useStylesScoped$,
+  JSXChildren,
+  useSignal,
+  $,
+  QwikChangeEvent,
+  PropFunction,
+  useResource$,
+  Resource,
+} from '@builder.io/qwik';
+import { PaginationParams } from '~/types/paginated';
+import { FilterInput } from './DefaultFilter';
+import styles from './table.css?inline';
+
+export interface TableProps<T extends string> {
+  rows: Record<T, JSXChildren>[];
+  customColumnNames?: Partial<Record<T, string>>;
+  total: number;
+  emitFetchRows: PropFunction<
+    ({ page, limit, filter, sort, sortColumn }: PaginationParams) => Promise<{
+      total: number;
+      data: Record<T, JSXChildren>[];
+    }>
+  >;
+}
+
+export const ServerPaginatedDataTable = component$(<T extends string>(props: TableProps<T>) => {
+  useStylesScoped$(styles);
+  const rowsPerPage = useSignal(10);
+  const currentPage = useSignal(0);
+  const filter = useSignal('');
+  const sortColumn = useSignal<keyof T>(Object.keys(props.rows[0])[0] as keyof T);
+  const sortDesc = useSignal(true);
+  const maxPages = useSignal(props.total);
+
+  const headers = Object.keys(props.rows[0]).map((colName) => ({
```

Comment:
Consider get the headers in a separate prop

Reply:
Props now include optional customColumnNames